src/sentry/integrations/github/client.py

```diff
@@ -62,20 +65,37 @@ def request(self, method, path, headers=None, data=None, params=None):
         return self._request(method, path, headers=headers, data=data, params=params)
 
     def get_token(self):
-        if not self.token or self.expires_at < datetime.datetime.utcnow():
+        """
+        Get token retrieves the active access token from the integration model.
+        Should the token have expried, a new token will be generated and
+        automatically presisted into the integration.
+        """
+        token = self.integration.metadata.get('access_token')
+        expires_at = self.integration.metadata.get('expires_at')
+
+        if expires_at is not None:
+            expires_at = datetime.strptime(expires_at, '%Y-%m-%dT%H:%M:%S')
+
+        if not token or expires_at < datetime.utcnow():
             res = self.create_token()
-            self.token = res['token']
-            self.expires_at = datetime.datetime.strptime(
+            token = res['token']
+            expires_at = datetime.strptime(
                 res['expires_at'],
                 '%Y-%m-%dT%H:%M:%SZ',
             )
 
-        return self.token
+            self.integration.metadata.update({
+                'access_token': token,
+                'expires_at': expires_at.isoformat(),
+            })
+            self.integration.save()
+
+        return token
 
     def create_token(self):
         return self.post(
             '/installations/{}/access_tokens'.format(
-                self.external_id,
+                self.integration.external_id,
```

Comment:
So  yeah this will be problematic for GHE, but we could easily just override the `create_token` method in the GHE client